### DISCUSSION TITLE
Grid Point Retrieval (Out of Date Content)

### DISCUSSION THREAD

**Discussion by grfritz** [weather-gov/api#380] [Q&A]
I have been initially trying out the API grid point forecast feature (i.e. https://api.weather.gov/gridpoints/IND/57,68) and discovered the forecast data spans time periods six or more hours in the past which have been superceded.  This is inconsistent with NDFD which consistently provides the forecast data beginning with the current time period.  This requires us to truncate the initial "obsolete" data returned by the API to avoid displaying forecasts for weather that has already occurred.

It would be great if the API point forecast functioned same as the NDFD and only returned current forecasts (instead of displaying time spans that have already been superceded).

Thanks

**Comment by StephenClouse** [comment]
The current behavior is by design and not likely to be changed.

Our design goal with API has been to represent the source data as accurately as possible, without modification (other than representational format conversion). What you are seeing is the forecast grid data precisely as we have received it from the field office.

What you might be wanting is the actual forecast (https://api.weather.gov/gridpoints/IND/57,68/forecast or https://api.weather.gov/gridpoints/IND/57,68/forecast/hourly) and not the raw data. The forecast endpoints do behave as you have specified.

**Comment by grfritz** [comment]
Thanks - unfortunately the forecast data I am using (qpf, snow, ice) only appear to be contained in the raw data.  If there is another way to access this via the API I would certainly be interested.  In any case, not a major issue filtering out the obsolete time periods.  Just wanted to confirm whether this was by intent.  Thanks.

**Comment by tmabell** [comment]
The simple solution to the many comments about this and the lack of QPF in the Forecast API could be solved by just adding QPF to the Forecast API.  The API is an extensive chuck of code, it can't be that hard to add one more element.  Funded or not someone needs to take some pride in their work.  Remember the name of the organization is The National Weather Service, with specific emphasis on the word "Service".
